### PR TITLE
Replace `grunt develop` and `grunt test` with Testem

### PR DESCRIPTION
**Pull request by patocallaghan**
Replaces `grunt-contrib-qunit` with `grunt-testem-mincer`. This allows us to drop support for PhantomJS and other dependencies like `grunt-contrib-connect` and `grunt-contrib-watch`.

It's worth noting I needed to use my own modified fork of `grunt-testem-mincer` as the version of testem it installed is pre 1.0 and this caused issues on CI.  It doesn't support sending flags to the test browser so we couldn't run Chrome on Travis CI without the sandbox. This meant Travis CI would fail with permission errors. Forking `grunt-testem-mincer` and updating Testem unblocks us as it gets tests passing on CI for the gj/2.12 branch (https://travis-ci.org/ebryn/ember-model/builds/3421447940). I have a PR open to update the Testem version in grunt-testem-mincer https://github.com/inossidabile/grunt-testem-mincer/pull/39.

/cc @GavinJoyce

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,22 @@
 language: node_js
 node_js:
   - "6"
```
Comment (patocallaghan) [inline .travis.yml]: I've removed one of the node versions as this is what was causing logs to never show on Travis.

File: package.json [hunk 2]
```diff
@@ -15,23 +15,22 @@
   "author": "no",
   "license": "MIT",
   "devDependencies": {
+    "glob": "~3.2.1",
     "grunt": "~1.0.1",
-    "grunt-neuter": "~0.6.0",
```
Comment (patocallaghan) [inline package.json]: It looks like this is mainly re-ordering of the packages.

**Comment by GavinJoyce** [comment]
👍 nice one @patocallaghan

File: .travis.yml (hunk 1 above)
Comment (GavinJoyce) [reply .travis.yml]: Node 8 is currently the LTS version. Perhaps we should remove `6` instead of `8`?

Comment (patocallaghan) [reply .travis.yml]: So the ember-cli adding .travis.yml actually recommends 4 but that fails to install due to peer dependencies not being met.

**Comment by GavinJoyce** [comment]
screen shot 2018-02-16 at 08 46 12

File: .travis.yml (hunk 1 above)
Comment (GavinJoyce) [reply .travis.yml]: Let's merge this now and come back to it later in https://github.com/ebryn/ember-model/issues/472

**Comment by GavinJoyce** [comment]
Merging now so that we have logs on CI, tracking a further upgrade to node in https://github.com/ebryn/ember-model/issues/472

**Pull request merged**